src/vellum/workflows/runner/runner.py

```diff
@@ -504,9 +496,16 @@ def _stream(self) -> None:
         for edge in self.workflow.get_edges():
             self._dependencies[edge.to_node].add(edge.from_port.node_class)
 
+        current_parent = WorkflowParentContext(
+            span_id=self._initial_state.meta.span_id,
+            workflow_definition=self.workflow.__class__,
+            parent=self._parent_context,
+            type="WORKFLOW",
+        )
         for node_cls in self._entrypoints:
             try:
-                self._run_node_if_ready(self._initial_state, node_cls)
+                with execution_context(parent_context=current_parent):
+                    self._run_node_if_ready(self._initial_state, node_cls)
```

Comment:
[nit] could see us doing the same strategy as `_context_run_work_item` to avoid all of the repeated `with .. as` statements